replication-api-impl/src/test/java/com/connexta/replication/api/impl/worker/WorkerManagerComponentTest.java

```diff
@@ -0,0 +1,160 @@
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.data.solr.repository.config.EnableSolrRepositories;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@ComponentScan(
+    basePackageClasses = {
+      WorkerManager.class,
+      ComponentTestConfig.class,
+      FilterIndexRepository.class,
+      EmbeddedSolrConfig.class
+    })
+@EnableSolrRepositories(basePackages = "com.connexta.replication")
+@RunWith(SpringRunner.class)
+public class WorkerManagerComponentTest {
+
+  private static final String LOCAL_SITE_ID = "localSiteId";
+  private static final String REMOTE_SITE_ID = "remoteSiteId";
+  private static final int QUEUE_CAPACITY = 20;
+
+  private static final MeterRegistry REGISTRY = new MeterRegistryMock();
+
+  private WorkerManager workerManager;
+
+  @Autowired private SiteManager siteManager;
+
+  @Rule public final MultiThreadedErrorCollector collector = new MultiThreadedErrorCollector();
+
+  private NodeAdapters nodeAdapters = mock(NodeAdapters.class);
+
+  private NodeAdapterFactory adapterFactory = mock(NodeAdapterFactory.class);
+
+  private QueueBroker broker;
+
+  private final NodeAdapter remote = mock(NodeAdapter.class);
+
+  @Before
+  public void setup() {
+    when(adapterFactory.create(any(URL.class))).thenReturn(remote);
+    when(nodeAdapters.factoryFor(SiteType.DDF)).thenReturn(adapterFactory);
+    when(remote.isAvailable()).thenReturn(true);
+    when(remote.createRequest(any(CreateRequest.class))).thenReturn(true);
+
+    broker =
+        new MemoryQueueBroker(
+            WorkerManagerComponentTest.QUEUE_CAPACITY, WorkerManagerComponentTest.REGISTRY);
+
+    ReplicationProperties props = new ReplicationProperties();
+    props.setLocalSite(LOCAL_SITE_ID);
+
+    final Site localSite = new SiteImpl(createSitePojo(LOCAL_SITE_ID, 1));
+    siteManager.save(localSite);
+
+    final Site remoteSite = new SiteImpl(createSitePojo(REMOTE_SITE_ID, 2));
+    siteManager.save(remoteSite);
+
+    workerManager = new WorkerManager(broker, siteManager, nodeAdapters, props);
+  }
+
+  @After
+  public void cleanup() throws Exception {
+    workerManager.destroy();
+    siteManager.objects().map(Site::getId).forEach(siteManager::remove);
+  }
+
+  @Test
+  public void testWorkerTakesFromQueue() throws Exception {
+    MemorySiteQueue queue = (MemorySiteQueue) broker.getQueue(REMOTE_SITE_ID);
+    TaskInfo taskInfo1 = createTaskInfo("taskId1");
+    TaskInfo taskInfo2 = createTaskInfo("taskId2");
+    TaskInfo taskInfo3 = createTaskInfo("taskId3");
+    queue.put(taskInfo1);
+    queue.put(taskInfo2);
+    queue.put(taskInfo3);
+
+    workerManager.init();
+
+    assertTimeoutPreemptively(
+        Duration.ofSeconds(10),
+        () -> {
+          while (true) {
+            if (queue.size() == 0) {
+              break;
+            }
+          }
+        });
+  }
```

Comment:
❗️ Missing validation of outcome from your component. Like the fact that the tasks should be completed, failed, or re-queued. Like the fact that data in the tasks should be sent out to the destination site.